app/src/androidTest/java/com/github/fribourgsdp/radio/WorkingJoinGameActivityTest.kt

```diff
@@ -47,42 +56,113 @@ class WorkingJoinGameActivityTest {
-        )
-
-        joinButton.perform(ViewActions.click())
-
-        Intents.intended(
-            Matchers.allOf(
-                IntentMatchers.toPackage("com.github.fribourgsdp.radio"),
-                IntentMatchers.hasComponent(LobbyActivity::class.java.name),
-                IntentMatchers.hasExtra(GAME_HOST_NAME_KEY, LocalDatabase.EXPECTED_SETTINGS.hostName),
-                IntentMatchers.hasExtra(GAME_NAME_KEY, LocalDatabase.EXPECTED_SETTINGS.name),
-                IntentMatchers.hasExtra(GAME_PLAYLIST_NAME_KEY,LocalDatabase.EXPECTED_SETTINGS.playlistName),
-                IntentMatchers.hasExtra(GAME_NB_ROUNDS_KEY, LocalDatabase.EXPECTED_SETTINGS.nbRounds),
-                IntentMatchers.hasExtra(GAME_HINT_KEY, LocalDatabase.EXPECTED_SETTINGS.withHint),
-                IntentMatchers.hasExtra(GAME_PRIVACY_KEY, LocalDatabase.EXPECTED_SETTINGS.isPrivate),
-                IntentMatchers.hasExtra(GAME_IS_HOST_KEY, false),
-                IntentMatchers.hasExtra(GAME_UID_KEY, testUID)
 
+            joinButton.perform(ViewActions.click())
+
+            Intents.intended(
+                Matchers.allOf(
+                    IntentMatchers.toPackage("com.github.fribourgsdp.radio"),
+                    IntentMatchers.hasComponent(LobbyActivity::class.java.name),
+                    IntentMatchers.hasExtra(GAME_HOST_NAME_KEY, LocalDatabase.EXPECTED_SETTINGS.hostName),
+                    IntentMatchers.hasExtra(GAME_NAME_KEY, LocalDatabase.EXPECTED_SETTINGS.name),
+                    IntentMatchers.hasExtra(GAME_PLAYLIST_NAME_KEY,LocalDatabase.EXPECTED_SETTINGS.playlistName),
+                    IntentMatchers.hasExtra(GAME_NB_ROUNDS_KEY, LocalDatabase.EXPECTED_SETTINGS.nbRounds),
+                    IntentMatchers.hasExtra(GAME_HINT_KEY, LocalDatabase.EXPECTED_SETTINGS.withHint),
+                    IntentMatchers.hasExtra(GAME_PRIVACY_KEY, LocalDatabase.EXPECTED_SETTINGS.isPrivate),
+                    IntentMatchers.hasExtra(GAME_IS_HOST_KEY, false),
+                    IntentMatchers.hasExtra(GAME_UID_KEY, testUID)
+                )
+            )
+        }
+    }
+
+
+    @Test
+    fun cancelScanQRCodeWork(){
+
+        val context: Context = ApplicationProvider.getApplicationContext()
+        val intent = Intent(context, QRCodeJoinGameActivity::class.java)
+        ActivityScenario.launch<QRCodeJoinGameActivity>(intent).use { scenario ->
+            ViewActions.closeSoftKeyboard()
+            val joinQRCodeButton = Espresso.onView(ViewMatchers.withId(R.id.joinWithQRCode))
+            joinQRCodeButton.perform(ViewActions.click())
+
+            Espresso.onView(ViewMatchers.withId(R.id.cancel_button))
+                .inRoot(RootMatchers.isDialog())
+                .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+
+            Espresso.onView(ViewMatchers.withId(R.id.cancel_button))
+                .inRoot(RootMatchers.isDialog())
+                .perform(ViewActions.click())
+        }
```

Comment:
Don't you want to test that what happens when you click on cancel works correctly?

Reply:
yes indeed an oversight on my part, well done